pkg/osbuild/chrony_stage.go

```diff
@@ -66,6 +69,65 @@ type ChronyConfigRefclock struct {
 	Offset *float64       `json:"offset,omitempty"`
 }
 
+type chronyConfigRefclockData struct {
+	Driver *json.RawMessage
+
+	Poll   *int     `json:"poll,omitempty"`
+	Dpoll  *int     `json:"dpoll,omitempty"`
+	Offset *float64 `json:"offset,omitempty"`
+}
+
+func (c *ChronyConfigRefclock) UnmarshalJSON(data []byte) (err error) {
+	var d chronyConfigRefclockData
+	if err := json.Unmarshal(data, &d); err != nil {
+		return err
+	}
+	c.Poll = d.Poll
+	c.Dpoll = d.Dpoll
+	c.Offset = d.Offset
+
+	var peek struct {
+		Name string
+	}
+	if err := json.Unmarshal(*d.Driver, &peek); err != nil {
+		return err
+	}
+	switch peek.Name {
+	case "PPS":
```

Comment:
I wonder if this should be case-insensitive? The docs use upper case for these, but users don't always follow directions. Do you happen to know what chrony does if they are lower case?

Reply:
Nevermind -- chronyd insists on upper case names :)